core/src/nucleus/ribosome/api/call.rs

```diff
@@ -112,38 +114,22 @@ pub(crate) fn reduce_call(
         return;
     }
     let dna = state.dna.clone().unwrap();
-    let maybe_cap = get_capability_with_zome_call(&dna, &fn_call);
-    if let Err(fn_res) = maybe_cap {
+    let maybe_public = is_fn_public(&dna, &fn_call);
+    if let Err(fn_res) = maybe_public {
         // Notify failure
         state
             .zome_calls
             .insert(fn_call.clone(), Some(fn_res.result()));
         return;
     }
-    let cap = maybe_cap.unwrap().clone();
+    let public = maybe_public.unwrap();
 
     // 2. Checks for permission to access Capability
-    // TODO #301 - Do real Capability token check
-    let can_call = match cap.cap_type.membrane {
-        Membrane::Public => true,
-        Membrane::Zome => {
-            // TODO #301 - check if caller zome_name is same as called zome_name
-            false
-        }
-        Membrane::Agent => {
-            // TODO #301 - check if caller has Agent Capability
-            false
-        }
-        Membrane::ApiKey => {
-            // TODO #301 - check if caller has ApiKey Capability
-            false
-        }
-    };
-    if !can_call {
+    if !public && !check_capability(context.clone(), &fn_call.clone()) {
```

Comment:
Ah I guess my comment above gets resolved by this: guess `check_cabapility` will be true for a local call?